src/formatDate.js

```diff
@@ -50,7 +50,63 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // verifying icoming formats
+  if (fromFormat.length !== 4) {
+    throw new Error('Incorrect date format (fromFormat).');
+  }
+
+  if (toFormat.length !== 4) {
+    throw new Error('Incorrect date format (fromFormat).');
+  }
+
+  // an object to neatly store all the date values
+  const dateInfo = {};
+  // separators
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
```

Comment:
I would change the naming of variables from `oldSeparator` and `newSeparator` to `fromSeparator` and `toSeparator`.